R/accessory.R

```diff
@@ -1091,6 +1104,121 @@ compile_contrast_data <-
     contrast_stats_rearranged
   }
 
+#' Validate design matrix and contrast specifications.
+#'
+#' This function performs several checks on the design matrix constructed
+#' from sample metadata.
+#'
+#' Checks:
+#' 1. Columns from design matrix are present in sample metadata.
+#' 2. Design matrix is full rank.
+#' 3. Contrasts are valid (i.e. specified coefficients occur as columns in the design matrix).
+#' 4. No NAs in any of the columns used in the design.
+#' 5. Warn about continuous covariates (e.g. numeric patient IDs treated as continuous).
+#' 6. Values of specified columns don't contain special characters that the pipeline can't handle.
+#'
+#' @param sample_metadata Data frame of sample metadata.
+#' @param design_cols Character vector of column names to be used for the design matrix.
+#' @param contrasts Optional list of contrasts. Each contrast should be a list with at least an 'id' and a 'comparison' field (a character vector).
+#' @param continuous_warning Optional character vector of column names that should be categorical.
+#' @param special_chars Character vector of special characters to check for (default: c("/", "\\\\")). Note: "\\" is escaped.
+#'
+#' @return TRUE if all checks pass. Stops with an error if a check fails.
+#' @export
+validate_design_matrix <- function(sample_metadata,
```

Comment:
Some of what is done here is already covered within read_contrasts. The new parts seem to be:

1. Checking for full rank
2. Checking continuous
3. Checking special characters

Those checks should be called along with the others in read_contrasts